README.md

```diff
@@ -1 +1,178 @@
+                return nil, errors.New("can't convert")
+            }
+            return iVal + 1, nil
+        }),
+})
+result, err := mapper.Map([]string{"a", "4"}) // []any{"a1", 5} <nil>
+
+result, err = mapper.MapWithErrHandler([]string{"a", "b"}, func(i int, err error) (any, error) {
+        return "null", nil
+}) // []any{"a1", "null"} <nil>
+```
+```golang
+// Single mapper example.
+toStringMapper := tupleconv.NewSingleMapper[any, string](tupleconv.MakeFuncConverter(
+    func(s any) (string, error) {
+        return fmt.Sprintln(s), nil
+    }),
+)
+res, err := toStringMapper.Map([]any{1, 2.5, nil}) // ["1\n", "2.5\n", "<nil>\n"] <nil>
+```
+**Observation 1**: To create a mapper, an array of converters is needed, each
+of which transforms a certain type S into type T.   
+
+**Observation 2**: To perform tuple mapping, you can use the function 
+`Map`, which will return control to the calling code upon the first error.  
+
+**Observation 3**: To perform tuple conversion, you can use the function 
+`MapWithErrHandler`. Upon a conversion error, it will call the function
+provided as the second argument, referred to as the Error Handler. 
+It will pass the index where the error occurred and the error itself to the 
+Error Handler. The Error Handler can return a new value for that index or 
+return an error again to halt the execution of the function at that point.
+
+**Observation 4**: You can use a single converter to create a mapper. 
+This mapper is created by the function `NewSingleMapper`, and its 
+`Map` function will work like a `map` function in functional languages.
+
+### Mappers to tarantool types
+
+#### Example
+For building an array of converters, especially when it comes to conversions to 
+tarantool types, there is a built-in solution.  
+Let's consider an example:
+```golang
+factory := tupleconv.NewStringToTTConvFactory().
+	WithDecimalSeparators(",.")
+
+spaceFmt := []tarantool.Field{
+    {Type: "unsigned"},
+    {Type: "double", IsNullable: true},
+    {Type: "string"},
+}
+
+converters, _ := tupleconv.MakeTypeToTTConverters[string](factory, spaceFmt)
+mapper := tupleconv.NewMapper(converters)
+result, err := mapper.Map([]string{"1", "-2,2", "some_string"}) // [1, -2.2, "some_string"] <nil>
+```
+**Observation 1**: To build an array of converters, the space format and a 
+certain object implementing `TypeToTTConvFactory` are used. Function 
+`MakeTypeToTTConverters` takes these entities and gives the converters list.
+
+**Observation 2**: `TypeToTTConvFactory[Type]` is capable of building a 
+converter from `Type` to each tarantool type.
+
+**Observation 3**: There is a basic factory available called 
+`StringToTTConvFactory`, which is used for conversions from strings to 
+tarantool types.   
+
+**Observation 4**: `StringToTTMapperFactory` can be configured with options like
+`WithDecimalSeparators`.
+
+#### String to nullable
+When converting nullable types with `StringToTTConvFactory`, first, an attempt
```

Comment:
What is a "nullable type" in Go?

Reply:
Here I meant tarantool types, not Go, as this sub-paragraph is part of the section about tarantool types.